api_client/python/timesketch_api_client/aggregation.py

```diff
@@ -272,3 +276,172 @@ def generate_chart(self):
 
         vega_spec_string = json.dumps(vega_spec)
         return altair.Chart.from_json(vega_spec_string)
+
+
+class AggregationGroup(resource.BaseResource):
+    """Aggregation Group object.
+
+    Attributes:
+        id: the ID of the group.
+    """
+
+    def __init__(self, sketch, api):
+        """Initialize the aggregation group."""
+        resource_uri = 'sketches/{0:d}/aggregation/group/'.format(
+            sketch.id)
+        super(AggregationGroup, self).__init__(api, resource_uri)
+
+        self.id = None
+        self._name = 'N/A'
+        self._description = 'N/A'
+        self._how = ''
+        self._parameters = {}
+        self._sketch = sketch
+        self._aggregations = []
+
+    def __str__(self):
+        """Return a string representation of the group."""
+        return '[{0:d}] {1:s} - {2:s}'.format(
+            self.id, self._name, self._description)
+
+    @property
+    def chart(self):
+        """Property that returns an altair Vega-lite chart."""
+        if not self._aggregations:
+            return altair.Chart()
+        return self.generate_chart()
+
+    @property
+    def description(self):
```

Comment:
These just returns the properties. Can we make them public and remove these?

E.g. _description --> description

Reply:
so I prefer not to do that, since I will be doing a follow up CL where I add a setter to this function, which will trigger an update of the database as soon as you change the parameter... for now, this makes this a read-only parameter... if I expose it with `.description` then it becomes a public parameter, meaning read/write and nothing changes if I modify it.